Commonalities/documentation/Working/user-consent.md

```diff
@@ -0,0 +1,396 @@
+}
+```
+
+The response must return a new created `access_token` with the granted scopes and purposes. The `authorization_id` in the response MUST be the same as the one included in the assertion. A new `authorization_id` MUST NOT BE created. New `authorization_id` can only be created using the Login authorization flow because it represents an access session for an authenticated user.
+
+The login authorization session expiration will be extended if the initial `authorization_id` was requested with the `offline_access` scope. The `refresh_token` obtained with initial `authorization_id` is still valid and can still be used to extend the login authorization expiration with the Refresh token.
+
+The Auth server MUST check the user consent of the purposes given in the assertion using the **Consent API**, verifying that the opt-in has been previously captured and registered by the application or a delegated third party. No error response will be received if the consent for non-automatic purposes is not active or revoked. The purposes will be just removed from the response or the access token. An error will only be received if after all the filtering processes in Auth Server there is no scope left in the access token.
+
+The assertion to request an access token related to a previous authorization will include de following fields:
+
+| **assertion field** | Description | Type |
+| --- | --- | --- |
+| `aud` | The audience for this assertion. | REQUIRED |
+| `iss` | The issuer identifier associated to the app. | REQUIRED |
+| `sub` | The sub that can be extracted from the ```id_token``` obtained with the Login authorization flow corresponding with the ```authorization_id```. It usually represents a user id. | REQUIRED |
+| `exp`| The timestamp (UNIX epoch time format) at which this JWT assertion will expire. | REQUIRED |
+| `iat` | The timestamp (UNIX epoch time format) at which this JWT assertion was issued. | REQUIRED |
+| `jti` | A globally unique identifier for this JWT assertion. | REQUIRED |
+| `authorization_id` | The `authorization_id` of the previous authentication obtained with Login authorization flow. It represents a valid authorization session for which the new access token will be created. | REQUIRED |
+| `scope`| The list of scopes, separated by space, that the app requests for the new access token. | OPTIONAL |
+| `purpose` | The list of purposes, separated by space, that the app requests for the new access token. | OPTIONAL |
+
+_Example: assertion to get an access token for a previous _authorization\_id_ for the non-automatic purpose "improve-network-performance"._
+
+```json
+{
+  "aud": "https://auth.example.com/",
+  "iss": "https://your-app-issuer.example.com",
+  "sub": "412d606f-4937-443b-b5e7-a8d0f63ef0bc",
+  "authorization_id": "96eee0c3-a4a0-4413-a9bd-3c30050c50d2",
+  "purpose": "improve-network-performance",
+  "exp": 1504807731,
+  "iat": 1504804131,
+  "jti": "53f42eb1-b751-44b5-bada-6990e08f35ad"
+}
```

Comment:
actor token not needed?

Reply:
For auth_code + jwt-bearer approach, the JWT assertion to request the access token to access the network resource mainly requires: 
- the `sub` (representing the user that have been authenticated in a previous Login Authorization flow... i.e. the one that granted the consent)
- authorization_id: repsenting the existing authorization session for that user.
- purpose: the purpose being requested for the access token.

With that information the Auth Server can validate (before issuing the access_token) that user was properly authenticated and validate if user consent exists for the required purpose.